bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/discovery/NetworkDiscoveryService.java

```diff
@@ -205,23 +213,25 @@ protected void startScan() {
     }
 
     @Override
-    protected synchronized void stopScan() {
-        super.stopScan();
-        final ExecutorService service = executorService;
-        if (service == null) {
-            return;
+    protected void stopScan() {
```

Comment:
I see now that my attempt to do the shutdown outside synchronization (to not delay everything else) is probably futile, because `AbstractDiscoveryService` calls it while holding a lock on the class... sigh. But, it won't make things worse, it just won't have any benefit.

The other changes (`shutdownNow()` etc.) should still help.